MediaBrowser.Controller/LiveTv/IListingsManager.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.LiveTv;
+
+namespace MediaBrowser.Controller.LiveTv;
+
+/// <summary>
+/// Service responsible for managing <see cref="IListingsProvider"/>s and mapping
+/// their channels to channels provided by <see cref="ITunerHost"/>s.
+/// </summary>
+public interface IListingsManager
+{
+    /// <summary>
+    /// Saves the listing provider.
+    /// </summary>
+    /// <param name="info">The listing provider information.</param>
+    /// <param name="validateLogin">A value indicating whether to validate login.</param>
+    /// <param name="validateListings">A value indicating whether to validate listings..</param>
+    /// <returns>Task.</returns>
+    Task<ListingsProviderInfo> SaveListingProvider(ListingsProviderInfo info, bool validateLogin, bool validateListings);
```

Comment:
Should we append Async to these function names while we're moving them?

Reply:
Generally I don't add the Async suffix unless there's a sync overload, but if we want to establish that as a convention we can do that.

Reply:
IMO we should, makes it easier to see when a Task isn't properly awaited during code review

Reply:
Do analyzers/warnings not cover that already? I would expect build to fail if someone didn't await a method like that

Reply:
They should, but in the past things have slipped through

Reply:
Alright, in that case I have one more queued PR for this sort of refactoring, then I can make a PR that applies that suffix everywhere in LiveTV